src/routes/moderation.rs

```diff
@@ -0,0 +1,90 @@
+use super::ApiError;
+use crate::auth::check_is_moderator_from_headers;
+use crate::database;
+use crate::models;
+use crate::models::mods::{ModStatus, VersionType};
+use actix_web::{get, web, HttpRequest, HttpResponse};
+use sqlx::PgPool;
+
+#[get("mods")]
+pub async fn mods(req: HttpRequest, pool: web::Data<PgPool>) -> Result<HttpResponse, ApiError> {
+    check_is_moderator_from_headers(req.headers(), &**pool).await?;
+
+    use futures::stream::TryStreamExt;
+
+    let mods = sqlx::query!(
+        "
+        SELECT * FROM mods
+        WHERE status = (
+            SELECT id FROM statuses WHERE status = $1
+        )
+        ORDER BY updated ASC
+        LIMIT 100;
+        ",
+        ModStatus::Processing.as_str()
+    )
+    .fetch_many(&**pool)
+    .try_filter_map(|e| async {
+        Ok(e.right().map(|m| models::mods::Mod {
+            id: database::models::ids::ModId(m.id).into(),
+            team: database::models::ids::TeamId(m.team_id).into(),
+            title: m.title,
+            description: m.description,
+            body_url: m.body_url,
+            published: m.published,
+            categories: vec![],
+            versions: vec![],
+            icon_url: m.icon_url,
+            issues_url: m.issues_url,
+            source_url: m.source_url,
+            status: ModStatus::Processing,
+            updated: m.updated,
+            downloads: m.downloads as u32,
+            wiki_url: m.wiki_url,
+        }))
+    })
+    .try_collect::<Vec<models::mods::Mod>>()
+    .await
+    .map_err(|e| ApiError::DatabaseError(e.into()))?;
+
+    Ok(HttpResponse::Ok().json(mods))
+}
+
+/// Returns a list of versions that need to be approved
+#[get("versions")]
+pub async fn versions(req: HttpRequest, pool: web::Data<PgPool>) -> Result<HttpResponse, ApiError> {
+    check_is_moderator_from_headers(req.headers(), &**pool).await?;
+
+    use futures::stream::TryStreamExt;
+
+    let versions = sqlx::query!(
+        "
+        SELECT * FROM versions
+        WHERE accepted = FALSE
+        LIMIT 100;
```

Comment:
This should be ordered by the creation date, oldest first.  The limit should also be configurable (we may also want pagination, but that's a bit more work and doesn't seem necessary in this case).